FirebaseCombineSwift/Sources/Auth/Auth+Combine.swift

```diff
@@ -0,0 +1,485 @@
+    ///
+    /// The publisher will emit events on the **main** thread.
+    ///
+    /// - Parameters:
+    ///   - email: The user's email address.
+    ///   - link: The email sign-in link.
+    /// - Returns: A publisher that emits the result of the sign in flow.
+    /// - Remark:
+    ///   Possible error codes:
+    ///   - `AuthErrorCodeOperationNotAllowed` - Indicates that email and password
+    ///     accounts are not enabled. Enable them in the Auth section of the
+    ///     Firebase console.
+    ///   - `AuthErrorCodeUserDisabled` - Indicates the user's account is disabled.
+    ///   - `AuthErrorCodeInvalidEmail` - Indicates the email address is malformed.
+    ///
+    ///   See `AuthErrors` for a list of error codes that are common to all API methods
+    @discardableResult
+    public func signIn(withEmail email: String,
+                       link: String) -> Future<AuthDataResult, Error> {
+      Future<AuthDataResult, Error> { promise in
+        self.signIn(withEmail: email, link: link) { authDataResult, error in
+          if let error = error {
+            promise(.failure(error))
+          } else if let authDataResult = authDataResult {
+            promise(.success(authDataResult))
+          }
+        }
+      }
+    }
+
+    /// Sends a sign in with email link to provided email address.
+    ///
+    /// The publisher will emit events on the **main** thread.
+    ///
+    /// - Parameters:
+    ///   - email: The email address of the user.
+    ///   - actionCodeSettings: An `ActionCodeSettings` object containing settings related to
+    ///     handling action codes.
+    /// - Returns: A publisher that emits whether the call was successful or not.
+    @discardableResult
+    public func sendSignInLink(toEmail email: String,
+                               actionCodeSettings: ActionCodeSettings) -> Future<Void, Error> {
+      Future<Void, Error> { promise in
+        self.sendSignInLink(toEmail: email, actionCodeSettings: actionCodeSettings) { error in
+          if let error = error {
+            promise(.failure(error))
+          } else {
+            promise(.success(()))
+          }
+        }
+      }
+    }
+
+    //  MARK: - Email-based Authentication Helpers
+
+    /// Fetches the list of all sign-in methods previously used for the provided email address.
+    ///
+    /// The publisher will emit events on the **main** thread.
+    ///
+    /// - Parameter email: The email address for which to obtain a list of sign-in methods.
+    /// - Returns: A publisher that emits a list of sign-in methods for the specified email
+    ///   address, or an error if one occurred.
+    /// - Remark: Possible error codes:
+    ///   - `AuthErrorCodeInvalidEmail` - Indicates the email address is malformed.
+    ///
+    ///   See `AuthErrors` for a list of error codes that are common to all API methods
+    public func fetchSignInMethods(forEmail email: String) -> Future<[String], Error> {
```

Comment:
Same here, does this and the ones below need @discardableResult?

Reply:
+1